File: backend/api/updateSpk.js
const authController = require('../controller/authController.js');
const express = require('express');
const router = express.Router();
const jwt = require('../utils/jwt.js');

router.post('/', jwt.verifyJWT, async(req, res) => {
    let decodedToken = req.decodedToken;
    let uid = decodedToken._uid;

    let deviceId = req.body.deviceId;
    let spkPub = JSON.parse(req.body.spkPub);
    let spkSig = JSON.parse(req.body.spkSig);

    console.log('🧧--------------------------');
    console.log(spkPub);
    console.log(spkSig);
    await authController.updateSpk(uid, spkPub, spkSig);
    console.log('🧧--------------------------');


    try {
        res.status(200).json({
            message: 'opk 上傳成功',
            data: null,
            token: null
        });
    }
    catch (err) {
        res.status(500).json({
            message: `後端發生例外錯誤： ${err.message}`,
            data: null,
            token: null
        });
    }
});

module.exports = router;

File: client/omelet/lib/api/post/update_spk.dart
import 'dart:convert';

import 'package:http/http.dart' as http;
import 'package:flutter_secure_storage/flutter_secure_storage.dart';

import './../../utils/server_uri.dart';

Future<http.Response> updateSpk(deviceId, spkPub, spkSig) async {
  const storage = FlutterSecureStorage();
  final token = await storage.read(key: 'token');
  final res = await http.post(Uri.parse('$serverUri/api/v1/update-spk'),
      headers: <String, String>{
        'Content-Type': 'application/json; charset=UTF-8',
        'Authorization': 'Bearer $token'
      },
      body: jsonEncode(<String, String>{
        'deviceId': deviceId,
        'spkPub': spkPub,
        'spkSig': spkSig,
      }));

  return res;
}
